### ISSUE TITLE
OpenAI chat completion API deprecated functions in favor of tools

### ISSUE THREAD

**Issue by edouard-blocktool** [vercel/ai#739]
### Feature Description

See https://platform.openai.com/docs/api-reference#chat-create-functions

`functions` is now deprecated and one should use `tools` instead. Note that this is not just cosmetic. Passing `tools` enable the execution of multiple function in parallel, as opposed to one function at a time.

`experimental_onFunctionCall` should probably be replaced with something to handle multiple tool calls.

### Use Case

Leverage new gpt4 turbo parallel function calling capabilities.

**Comment by edouard-blocktool** [comment]
Dupe of https://github.com/vercel/ai/issues/732

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **None**  
Confidence: **Medium**

### Rationale
Although the underlying request is for a new capability, this issue was identified as duplicate bookkeeping rather than a standalone feature request.

### Evidence
- The request proposes replacing deprecated `functions` handling with OpenAI `tools` support for parallel tool calls.
- The only follow-up states that the issue is a duplicate of #732.
